core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2665,8 +2665,19 @@ object SparkContext extends Logging {
     // When running locally, don't try to re-execute tasks on failure.
     val MAX_LOCAL_TASK_FAILURES = 1
 
+    val cpusPerTask = sc.conf.get(CPUS_PER_TASK)
+
+    def checkClusterExecutorCores(): Unit = {
```

Comment:
Can you just modify this little utility method to take a "cores" parameter and then use it in all the cases below? it can default to `sc.conf.get(EXECUTOR_CORES)`, and then below you can set it to 1 for the local case, for example.